benches/Cargo.toml

```diff
@@ -71,12 +71,12 @@ name = "sync_semaphore"
 path = "sync_semaphore.rs"
 harness = false
 
-[[bench]]
+[[target.'cfg(unix)'.bench]]
```

Comment:
Sorry if I'm missing something, but is this syntax actually supported?
On my environment (macos, rustc 1.87), cargo commands are emitting a warning.

```bash
$ cargo check
warning: /Users/mox692/work/tokio/benches/Cargo.toml: unused manifest key: target.cfg(unix).bench
```

Reply:
@vilgotf 
Do you want to submit a follow-up PR?
I guess it's fine to define another main function when `#![cfg(not(unix))]`

Reply:
I'm facing the same issue while developing locally.

```
$ cargo +1.77 clippy --all --tests --all-features
warning: /xxx/src/tokio/benches/Cargo.toml: unused manifest key: target.cfg(unix).bench
```

@vilgotf Would mind opening another PR to fix this issue?

Reply:
I've submitted a PR for this: #7412